lookup_handler.go

```diff
@@ -8,24 +8,27 @@ import (
 	"k8s.io/client-go/tools/cache"
 )
 
-// LookupHandler is a handler for objects that can be mapped to some parent object
-// through the use of an SharedIndexInformer.
+// LookupHandler is a handler for controlled objects that can be mapped to some controller object
+// through the use of a Lookup function.
+// This handler assumes that the Logger already has the ctrl_gk field set.
 type LookupHandler struct {
-	Logger       *zap.Logger
-	WorkQueue    WorkQueueProducer
-	ZapNameField ZapNameField
+	Logger    *zap.Logger
+	WorkQueue WorkQueueProducer
 
 	Lookup func(runtime.Object) ([]runtime.Object, error)
 }
 
-func (e *LookupHandler) enqueueMapped(logger *zap.Logger, obj meta_v1.Object) {
+func (e *LookupHandler) enqueueMapped(obj meta_v1.Object, addUpdateDelete string) {
+	logger := e.loggerForObj(obj)
 	objs, err := e.Lookup(obj.(runtime.Object))
 	if err != nil {
-		logger.Error("Failed to get objects from index", zap.Error(err))
+		logger.Error("Failed to lookup objects", zap.Error(err))
 		return
 	}
 	for _, o := range objs {
 		metaobj := o.(meta_v1.Object)
+		logger.With(logz.Controller(metaobj)).Sugar().
+			Infof("Enqueuing controller object because controlled object was %s", addUpdateDelete)
```

Comment:
This now logs for each object and does not log if there are no objects to enqueue.